cookbooks/creating_a_behat_extension.rst

```diff
@@ -0,0 +1,285 @@
+
+Directory structure:
+
+.. code-block::
+  
+  src/
+      Context/
+          LogsContext.php   # This is where we'll implement our logging logic
+
+
+The code for ``LogsContext.php``:
+
+.. code-block:: php
+
+  <?php
+  
+  namespace Behat\LogsExtension\Context;
+  
+  use Behat\Behat\Context\Context;
+  use Behat\Behat\Hook\Scope\BeforeScenarioScope;
+  use Behat\Behat\Hook\Scope\AfterScenarioScope;
+  
+  class LogsContext implements Context
+  {
+      /** @BeforeScenario */
+      public function before(BeforeScenarioScope $scope)
+      {
+          if (/* enable config */ === false) {
+              return;
+          }
+
+          file_put_contents(
+              /* filepath */,
+              'START: ' . $scope->getScenario()->getTitle() . ' - ' . time() . PHP_EOL,
+              FILE_APPEND
+          );
+      }
+  
+      /** @AfterScenario */
+      public function after(AfterScenarioScope $scope)
+      {
+          if (/* enable config */ === false) {
+              return;
+          }
+  
+          file_put_contents(
+              /* filepath */,
+              'END: ' . $scope->getScenario()->getTitle() . ' - ' . time() . PHP_EOL,
+              FILE_APPEND
+          );
+      }
+  }
+
+Creating the Extension
+----------------------
+
+Next, we need to create the extension itself.
+This will serve as the entry point for our logging functionality.
+
+Directory structure:
+
+.. code-block::
+  
+  src/
+      Context/
+          LogsContext.php
+      ServiceContainer/
+          LogsExtension.php   # This is where we'll define our extension
+
+To ensure Behat can find and load the ``LogsExtension.php`` file, it is important to place it within the ``ServiceContainer`` folder.
+While there might be alternatives, we will stick to the straightforward method.
+
+The ``getConfigKey`` method is used to identify our extension in the configuration, and the ``configure`` method is used to define the configuration tree.
+
+The code for ``LogsExtension.php``:
+
+.. code-block:: php
+  
+  <?php
+  
+  namespace Behat\LogsExtension\ServiceContainer;
+  
+  use Behat\Testwork\ServiceContainer\Extension;
+  use Behat\Testwork\ServiceContainer\ExtensionManager;
+  use Symfony\Component\Config\Definition\Builder\ArrayNodeDefinition;
+  use Symfony\Component\DependencyInjection\ContainerBuilder;
+  
+  class LogsExtension implements Extension
+  {
+      public function getConfigKey()
+      {
+          return 'logs_extension';
+      }
+  
+      public function initialize(ExtensionManager $extensionManager)
+      {
+          // Empty for our case, but useful to hook into other extensions' configurations
+      }
+
+      public function configure(ArrayNodeDefinition $builder)
+      {
+          $builder
+              ->addDefaultsIfNotSet()
+              ->children()
+                  ->scalarNode('enable')->defaultFalse()->end()
```

Comment:
this looks like a use case for a `booleanNode` instead

Reply:
Sure! Fixed: 8e8526805ae58c80b79ef17954b1c7dd6a9588be